openwisp_controller/subnet_division/tasks.py

```diff
@@ -116,12 +121,30 @@ def _create_ipaddress_and_subnetdivision_index_objects(ips, indexes):
 
 
 @shared_task
-def provision_subnet_ip_for_existing_devices(organization_id):
-    for config in (
-        Config.objects.select_related('device', 'device__organization')
-        .filter(device__organization_id=organization_id)
-        .iterator()
-    ):
-        DeviceSubnetDivisionRuleType.provision_receiver(
-            config, created=True,
+def provision_subnet_ip_for_existing_devices(rule_id, organization_id, type):
+    if 'device' in type:
+        for config in (
+            Config.objects.select_related('device', 'device__organization')
+            .filter(device__organization_id=organization_id)
+            .iterator()
+        ):
+            DeviceSubnetDivisionRuleType.provision_receiver(
+                config, created=True,
+            )
+    elif 'vpn' in type:
+        rule = SubnetDivisionRule.objects.get(id=rule_id)
+        organization_filter = Q(organization_id=organization_id) | Q(
+            organization_id=None
+        )
+        vpn_qs = (
+            Vpn.objects.filter(subnet=rule.master_subnet)
+            .filter(organization_filter)
+            .values_list('id')
         )
+        qs = VpnClient.objects.filter(
+            vpn__in=vpn_qs, config__device__organization_id=organization_id
+        )
+        for vpn_client in qs:
+            VpnSubnetDivisionRuleType.provision_receiver(
+                instance=vpn_client, created=True
+            )
```

Comment:
why don't we use a method on the rules here?

Eg:

```python
try:
    rule = SubnetDivisionRule.objects.get(id=rule_id)
except SubnetDivisionRule.DoesNotExist:
    # ... please always handle this case! See how we do this in other tasks of OpenWISP Controller.

rule.type_class.provision_existing(.....)
```

BTW it seems to me that the `SubnetDivisionRule` model is missing methods like:
https://github.com/openwisp/openwisp-controller/blob/ff9095b4af5648487a1d388ce9a38cd2c849cb30/openwisp_controller/config/base/base.py#L130-L142

I didn't notice this before. How does the current code load the right rule class and instantiate it?

Reply:
The rule class is never instantiated. `class` is used to structure the code. All of the methods are class methods. 
Adding a `backend_class` property will be useful here.

Reply:
Ok only `rule_class` then.